src/lineup/internal/cmds.ts

```diff
@@ -269,9 +269,16 @@ export class LineupTrackingManager {
     const p: LocalDataProvider = await ResolveNow.resolveImmediately((await inputs[0].v).data);
     const ranking = p.getRankings()[parameter.rid];
 
-    const waitForAggregated = LineupTrackingManager.getInstance().dirtyRankingWaiter(ranking);
     LineupTrackingManager.getInstance().ignoreNext = LocalDataProvider.EVENT_GROUP_AGGREGATION_CHANGED;
 
+    let resolver: () => void;
+    const waiter = new Promise<void>((resolve) => resolver = resolve);
+
+    p.on(`${LocalDataProvider.EVENT_GROUP_AGGREGATION_CHANGED}.track`, () => {
+      p.on(`${LocalDataProvider.EVENT_GROUP_AGGREGATION_CHANGED}.track`, null); // disable
+      resolver(); // resolve waiter promise
+    });
```

Comment:
Wouldn't it be possible to avoid the setting of the resolve callback and do everything *ìnline* instead?

```
const waiter = new Promise<void>((resolve) => {
   p.on(`${LocalDataProvider.EVENT_GROUP_AGGREGATION_CHANGED}.track`, () => {
      p.on(`${LocalDataProvider.EVENT_GROUP_AGGREGATION_CHANGED}.track`, null); // disable
      resolve(); // resolve promise
   });
});
```

Reply:
Yes, good suggestion. I was using the structure of the `dirtyRankingWaiter()` function. I will adapt the code.